lib/ecto/query/planner.ex

```diff
@@ -743,7 +743,7 @@ defmodule Ecto.Query.Planner do
   defp plan_combinations(query, adapter) do
     combinations =
       Enum.map query.combinations, fn {type, combination_query} ->
-        {prepared_query, _params, _key} = plan(combination_query, :all, adapter)
+        {prepared_query, _params, _key} = plan(%{combination_query | prefix: query.prefix}, :all, adapter)
```

Comment:
If a prefix is set, we need to respect it though. it should be something like this:

```elixir
defp attach_prefix(%{prefix: nil} = query, %{prefix: prefix}), do: %{query | prefix: prefix}
defp attach_prefix(query, _), do: %{query | prefix: prefix}
```

And we should be good to go!

Reply:
👍 
I've made a requested change here 68f7c5b1540d06102f08ef306c9c48314ceaf78b